Source/OxyPlot/Rendering/TextArranger.cs

```diff
@@ -0,0 +1,258 @@
+            }
+
+            // position the lines
+            linePositions = new ScreenPoint[lines.Length];
+            for (int i = 0; i < lines.Length; i++)
+            {
+                var line = lines[i];
+                var lineWidth = this.TextMeasurer.MeasureTextWidth(line, fontFamily, fontSize, fontWeight);
+
+                var offsetLineWidth = new ScreenVector(cos * lineWidth, sin * lineWidth);
+
+                var offsetLineX = offsetLineWidth * offsetLineXRelative;
+
+                linePositions[i] = p + ((offsetLineHeight + offsetLeading) * i) + offsetLineX + offsetLineY;
+            }
+        }
+
+        /// <summary>
+        /// Measures the size of the text as it would be when arranged by the ArrangeText method."/>.
+        /// </summary>
+        /// <param name="text">The text.</param>
+        /// <param name="fontFamily">The font family.</param>
+        /// <param name="fontSize">Size of the font (in device independent units, 1/96 inch).</param>
+        /// <param name="fontWeight">The font weight.</param>
+        /// <returns>The size of the text.</returns>
+        /// <remarks>Returns an <see cref="OxySize"/> with no width and height if the text is <c>null</c> or empty.</remarks>
+        public OxySize MeasureText(
+            string text,
+            string fontFamily,
+            double fontSize,
+            double fontWeight)
+        {
+            if (string.IsNullOrEmpty(text))
+            {
+                // It is a bit of a lie to do this here, but otherwise everyone else has to do it
+                return OxySize.Empty;
+            }
+
+            var lines = StringHelper.SplitLines(text);
+
+            var width = lines.Max(l => this.TextMeasurer.MeasureTextWidth(l, fontFamily, fontSize, fontWeight));
+
+            var metrics = this.TextMeasurer.GetFontMetrics(fontFamily, fontSize, fontWeight);
+
+            var lineHeight = metrics.LineHeight;
+            var leading = metrics.Leading;
+            var lineCount = lines.Length;
+
+            var height = ((lineHeight + leading) * lineCount) - leading;
+
+            return new OxySize(width, height);
+        }
+
+        /// <summary>
+        /// Translates a <see cref="HorizontalAlignment"/> into a relative offset.
+        /// </summary>
+        /// <param name="horizontalAlignment">The horizontal alignent.</param>
+        /// <returns>The offset.</returns>
+        /// <remarks>
+        /// Left   -> -0.5
+        /// Center ->  0.0
+        /// Right  -> +0.5
+        /// </remarks>
+        private static double ResolveOffset(HorizontalAlignment horizontalAlignment)
+        {
+            var offset =
+                horizontalAlignment == HorizontalAlignment.Left ? -0.5
+                : horizontalAlignment == HorizontalAlignment.Center ? 0.0
+                : horizontalAlignment == HorizontalAlignment.Right ? 0.5
+                : throw new ArgumentException(nameof(horizontalAlignment));
```

Comment:
Could this be a switch expression?
I know you don't like them too much, but I find this a bit hard to read ;).

Reply:
Shall do. As long as there is an explicit default I don't mind them; It's the implicit throw that irritates me (so I have a dim view of the design altogether).